src/v/datalake/translation/partition_translator.cc

```diff
@@ -107,12 +115,30 @@ class coordinator_table_creator : public table_creator {
     coordinator::frontend& coordinator_fe_;
 };
 
+ss::future<cluster::errc> wait_stm_translated(
+  ss::shared_ptr<translation_stm> stm,
+  model::offset o,
+  model::timeout_clock::time_point deadline,
+  std::optional<std::reference_wrapper<ss::abort_source>> as) {
+    try {
+        co_await stm->wait_translated(model::prev_offset(o), deadline, as);
+    } catch (const ss::abort_requested_exception&) {
+        co_return cluster::errc::shutting_down;
+    } catch (const ss::timed_out_error&) {
+        co_return cluster::errc::timeout;
+    }
+    co_return cluster::errc::success;
+}
+
 } // namespace
 
 static constexpr std::chrono::milliseconds translation_jitter{500};
 constexpr ::model::timeout_clock::duration wait_timeout = 5s;
 
-partition_translator::~partition_translator() = default;
+partition_translator::~partition_translator() {
+    _partition->unregister_flush_hook(_partition_flush_subscription);
+};
```

Comment:
should that be unregistered in `stop()` ?

Reply:
it does not seem to make any difference, as the hook function doesn't even need the object to remain live after its scheduling point. But I'll move it to `stop()` to make things neat.